speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -188,6 +250,64 @@ static void SaveSettings(void* /* dummy */)
     GfuiScreenActivate(s_nextHandle);
 }
 
+/// @brief                    Sets all the checkboxes and radiobuttons in the researcher menu
+void InitializeResearcherMenuButtons()
+{
+    GfuiCheckboxSetChecked(s_scrHandle, m_indicatorsGui[0], m_indicators.Audio);
+    GfuiCheckboxSetChecked(s_scrHandle, m_indicatorsGui[1], m_indicators.Icon);
+    GfuiCheckboxSetChecked(s_scrHandle, m_indicatorsGui[2], m_indicators.Text);
+    GfuiCheckboxSetChecked(s_scrHandle, m_pControlGui[0], m_pControl.ControlInterventionToggle);
+    GfuiCheckboxSetChecked(s_scrHandle, m_pControlGui[1], m_pControl.ControlGas);
+    GfuiCheckboxSetChecked(s_scrHandle, m_pControlGui[2], m_pControl.ControlSteering);
+    GfuiCheckboxSetChecked(s_scrHandle, m_pControlGui[3], m_pControl.ForceFeedback);
+    GfuiRadioButtonListSetSelected(s_scrHandle, m_taskGui, m_task);
+    GfuiRadioButtonListSetSelected(s_scrHandle, m_interventionTypeGui, m_interventionType);
+    char buf[32];
+    sprintf(buf, "%d", m_maxTime);
+    GfuiEditboxSetString(s_scrHandle, m_maxTimeControl, buf);
+}
+
+/// @brief         Initializes the menu setting from the ResearcherMenu.xml file
+/// @param p_param The configuration menu handle
+void InitializeResearcherMenuSettings(void* p_param)
+{
+    // Retrieve all setting variables from the xml file and assigning them to the internal variables
+    m_task = std::stoi(GfParmGetStr(p_param, RESEARCHMENU_TASKS, GFMNU_ATTR_SELECTED, NULL));
+
+    m_indicators.Audio = gfuiMenuGetBoolean(GfParmGetStr(p_param, RESEARCHMENU_INDICATOR_AUDITORY, GFMNU_ATTR_CHECKED, NULL), true);
```

Comment:
Use this function from `guimenu.cpp`. This is exactly what you're doing here, but abstracted into a single function call. Don't forget to update all the other getBooleans below as well.
```
static bool
getControlBoolean(void* hparm, const char* pszPath, const char* pszFieldName, bool bDefault)
{
    return gfuiMenuGetBoolean(GfParmGetStr(hparm, pszPath, pszFieldName, 0), bDefault);
}
```